db/migrate/20160317160230_move_user_roles_to_entitlements.rb

```diff
@@ -0,0 +1,16 @@
+class MoveUserRolesToEntitlements < ActiveRecord::Migration[5.0]
+  class MiqGroup < ActiveRecord::Base; end
+  class Entitlement < ActiveRecord::Base; end
+
+  def up
+    MiqGroup.find_each do |group|
+      Entitlement.create!(:miq_group_id => group.id, :miq_user_role_id => group.miq_user_role_id)
+    end
+  end
+
+  def down
+    MiqGroup.find_each do |group|
+      Entitlement.find_by(:miq_group_id => group.id).try(:delete)
+    end
```

Comment:
Delete because we don't care about callbacks (faster, but only in combo
with your previous comment to use .where)

On Thursday, March 17, 2016, Joe Rafaniello <notifications@github.com>
wrote:

> In db/migrate/20160317160230_move_user_roles_to_entitlements.rb
> <https://github.com/ManageIQ/manageiq/pull/7302#discussion_r56583510>:
>
> > @@ -0,0 +1,16 @@
> > +class MoveUserRolesToEntitlements < ActiveRecord::Migration[5.0]
> > +  class MiqGroup < ActiveRecord::Base; end
> > +  class Entitlement < ActiveRecord::Base; end
> > +
> > +  def up
> > +    MiqGroup.find_each do |group|
> > +      Entitlement.create!(:miq_group_id => group.id, :miq_user_role_id => group.miq_user_role_id)
> > +    end
> > +  end
> > +
> > +  def down
> > +    MiqGroup.find_each do |group|
> > +      Entitlement.find_by(:miq_group_id => group.id).try(:delete)
> > +    end
>
> Also, why the try? Why delete? Curious...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/7302/files/0e05d3294527d5fec98ea5435155d45a08e1ed77#r56583510>
>


-- 
Chris Arcand
@chrisarcand